pkg/operator/saptuneapplysolution_v1.go

```diff
@@ -81,123 +122,60 @@ func NewSaptuneApplySolution(
 }
 
 func (sa *SaptuneApplySolution) plan(ctx context.Context) error {
-	opArguments, err := parseSaptuneApplyArguments(sa.arguments)
+	opArguments, err := parseSaptuneSolutionArguments(sa.arguments)
 	if err != nil {
 		return err
 	}
 	sa.parsedArguments = opArguments
 
-	// check saptune version
-	versionOutput, err := sa.executor.Exec(ctx, "rpm", "-q", "--qf", "%{VERSION}", "saptune")
-	if err != nil {
-		return fmt.Errorf(
-			"could not get the installed saptune version: %w",
-			err,
-		)
-	}
-	sa.logger.Debugf("installed saptune version: %s", string(versionOutput))
-
-	if supported := isSaptuneVersionSupported(string(versionOutput)); !supported {
-		return fmt.Errorf(
-			"saptune version not supported, installed: %s, minimum supported: %s",
-			versionOutput,
-			minimalSaptuneVersion,
-		)
+	if err = sa.saptune.CheckVersionSupport(ctx); err != nil {
+		return err
 	}
 
-	solutionAppliedOutput, err := sa.executor.Exec(ctx, "saptune", "--format", "json", "solution", "applied")
+	initiallyAppliedSolution, err := sa.saptune.GetAppliedSolution(ctx)
 	if err != nil {
-		return errors.New("could not call saptune solution applied")
+		return err
 	}
 
-	sa.resources[beforeDiffField] = string(solutionAppliedOutput)
+	sa.resources[beforeDiffField] = initiallyAppliedSolution
 
 	return nil
 }
 
 func (sa *SaptuneApplySolution) commit(ctx context.Context) error {
-	// check if solution is already applied
-	solutionAppliedOutput, err := sa.executor.Exec(ctx, "saptune", "--format", "json", "solution", "applied")
-	if err != nil {
-		return errors.New("could not call saptune solution applied")
+	initiallyAppliedSolution, ok := sa.resources[beforeDiffField].(string)
```

Comment:
strange, i didn't need to do this in the cluster maintenance operator. 
maybe booleans and strings behave differently

Reply:
it can be skipped, but it might panic 🤷